cmd/osbuild-composer/composer.go

```diff
@@ -214,91 +216,93 @@ func (c *Composer) Start() error {
 		logrus.Fatal("neither the weldr API socket nor the composer API socket is enabled, osbuild-composer is useless without one of these APIs enabled")
 	}
 
+	localWorkerAPI := &http.Server{
+		ErrorLog: c.logger,
+		Handler:  c.workers.Handler(),
+	}
+
+	remoteWorkersHandler := c.workers.Handler()
+	if c.config.Worker.EnableJWT {
+		var err error
+		keysURLs := c.config.Worker.JWTKeysURLs
+		remoteWorkersHandler, err = auth.BuildJWTAuthHandler(
+			keysURLs,
+			c.config.Worker.JWTKeysCA,
+			c.config.Worker.JWTACLFile,
+			[]string{
+				"/api/image-builder-worker/v1/openapi/?$",
+			},
+			c.workers.Handler(),
+		)
+		if err != nil {
+			panic(err)
+		}
+	}
+	remoteWorkerAPI := &http.Server{
+		ErrorLog: c.logger,
+		Handler:  remoteWorkersHandler,
+	}
+
+	const apiRouteV2 = "/api/image-builder-composer/v2"
+	const kojiRoute = "/api/composer-koji/v1"
+
+	mux := http.NewServeMux()
+
+	// Add a "/" here, because http.ServeMux expects the
+	// trailing slash for rooted subtrees, whereas the
+	// handler functions don't.
+	mux.Handle(apiRouteV2+"/", c.api.V2(apiRouteV2))
+	mux.Handle(kojiRoute+"/", c.koji.Handler(kojiRoute))
+
+	// Metrics handler attached to api mux to avoid a
+	// separate listener/socket
+	mux.Handle("/metrics", promhttp.Handler().(http.HandlerFunc))
+
+	composerHandler := http.Handler(mux)
+	var err error
+	if c.config.Koji.EnableJWT {
+		keysURLs := c.config.Koji.JWTKeysURLs
+		composerHandler, err = auth.BuildJWTAuthHandler(
+			keysURLs,
+			c.config.Koji.JWTKeysCA,
+			c.config.Koji.JWTACLFile,
+			[]string{
+				"/api/image-builder-composer/v2/openapi/?$",
+				"/api/image-builder-composer/v2/errors/?$",
+				"/metrics/?$",
+			}, mux)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	composerAPI := &http.Server{
+		ErrorLog: c.logger,
+		Handler:  composerHandler,
+	}
+
 	if c.localWorkerListener != nil {
 		go func() {
-			s := &http.Server{
```

Comment:
I think I preferred to have the listener-specific bits in the conditions. It might be slightly cleaner to define all `http.Server`s outside the conditions, initialize them in these conditions and call `Serve` in goroutines. WDYT?

Reply:
Agreed. This should also make it easier to review. Fixed.